src/sagemaker/session.py

```diff
@@ -4114,6 +4126,21 @@ def submit(request):
                         ModelPackageGroupName=request["ModelPackageGroupName"]
                     )
                 )
+            if "SourceUri" in request and request["SourceUri"] is not None:
+                if re.match(MODEL_PACKAGE_ARN_PATTERN, request["SourceUri"]) or re.match(
+                    MODEL_ARN_PATTERN, request["SourceUri"]
+                ):
+                    if "InferenceSpecification" in request:
+                        del request["InferenceSpecification"]
+                    return self.sagemaker_client.create_model_package(**request)
```

Comment:
I believe you are doing this when source uri is model package or model then remove inference spec and only include sourceuri for auto-population. Can we add comment for the same.
Also instead of `if` condition lets add a utility function to see if sourceUri can be auto-populated. As of now it would be restricted to model package and model arn but in future more arns we can auto-populate if possible.